pkgs/applications/office/pyspread/default.nix

```diff
@@ -1,42 +1,46 @@
 { lib
-, buildPythonApplication
-, fetchPypi
+, python3
+, fetchpatch
 , makeDesktopItem
-, makePythonPath
-, dateutil
-, matplotlib
-, numpy
-, pyenchant
-, pyqt5
-, pytest
-, python
 , qtsvg
-, runtimeShell
 , wrapQtAppsHook
 }:
 
-buildPythonApplication rec {
+python3.pkgs.buildPythonApplication rec {
   pname = "pyspread";
   version = "1.99.5";
 
-  src = fetchPypi {
+  src = python3.pkgs.fetchPypi {
     inherit pname version;
     sha256 = "sha256-05bC+Uvx72FAh3qxkgXm8jdb/gHRv1D/M7tjOEdE3Xg=";
   };
 
-  pythonLibs = [
+  patches = [
+    # https://gitlab.com/pyspread/pyspread/-/merge_requests/34
+    (fetchpatch {
+      name = "entry-points.patch";
+      url = "https://gitlab.com/pyspread/pyspread/-/commit/3d8da6a7a7d76f7027d77ca95fac103961d729a2.patch";
+      excludes = [ "bin/pyspread" "bin/pyspread.bat" ];
+      sha256 = "1l614k7agv339hrin23jj7s1mq576vkdfkdim6wp224k7y37bnil";
+    })
+  ];
+
+  nativeBuildInputs = [
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    qtsvg
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
```

Comment:
Can someone explain me what is the rationale/standard here? 

Is it to use `python3.pkgs` or `python3Packages`?

It is to declare these packages explicitly at the input parameters, or to expose them via `with`?

Reply:
> Is it to use `python3.pkgs` or `python3Packages`?

This only matters if you also input python(3) which could get mixed python versions when used together with python3Packages.

> It is to declare these packages explicitly at the input parameters, or to expose them via with?

If you do not declare them explicitly they cannot be easily overwritten.

Reply:
> If you do not declare them explicitly they cannot be easily overwritten.

It's the exact opposite: You should override `python3` to override a python package. So `python3` should be the only argument.

Reply:
> > If you do not declare them explicitly they cannot be easily overwritten.
> 
> It's the exact opposite: You should override `python3` to override a python package. So `python3` should be the only argument.

But Python is not what the programs wait, but Python libraries.

Reply:
That's what I mean with python package. If you want to override `python3.pkgs.xyz`, you need to use `python3.override { packageOverrides = ... }`.